obspy/core/tests/test_stream.py

```diff
@@ -2484,6 +2484,26 @@ def test_passing_kwargs_to_trace_detrend(self):
             self.assertEqual(arg[1]["order"], 2)
             self.assertEqual(arg[1]["plot"], True)
 
+    def test_read_no_check_compression(self):
+        """
+        Test to ensure calling read with check_compression=False does not
+        call expensive tar or zip functions.
+        """
+        # write a file to disk
+        file_name = 'temp.mseed'
+        st = read()
+        st.write(file_name, 'mseed')  # maybe read from test data instead?
+
+        with mock.patch("tarfile.is_tarfile") as tar_patch:
+            with mock.patch("zipfile.is_zipfile") as zip_patch:
+                read(file_name, check_compression=False)
```

Comment:
You could also add the opposite test, i.e. that the unzip check routines get called if `check_compression` is not overridden.